Source/WebCore/platform/mediarecorder/MediaRecorderPrivateEncoder.cpp

```diff
@@ -227,15 +229,16 @@ void MediaRecorderPrivateEncoder::resume()
 {
     assertIsMainThread();
 
+    if (!hasAudio())
+        m_resumeWallTime = MonotonicTime::now();
```

Comment:
It is a bit odd to set `m_resumeWallTime` here and in `appendVideoFrame` as well.
Should we add a `!m_resumeWallTime.load()` check as well?

Reply:
This is from the original MediaRecorder implementation (that you wrote :) ). It is only used to determine the duration of the video. In `appendVideoFrame` it deals on setting it for the first time ever, which will be used to set the time of the first frame.

When we pause, `now()-m_resumeWallTime` becomes the duration of the video and `m_resumeWallTime` is then set upon the next resume so that we can determine the duration of the new video segment.

I could probably makes this code a bit easier to follow.

Reply:
I've reworked things a bit so that it's now all done in the workQueue and renamed the member to `m_currentVideoSegmentStartTime` .